x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/use_create_analytics_form.ts

```diff
@@ -125,25 +125,68 @@ export const useCreateAnalyticsForm = (): CreateAnalyticsFormProps => {
 
   const createKibanaIndexPattern = async () => {
     const dataViewName = destinationIndex;
+    let retryCount = 15;
 
     try {
-      await mlContext.dataViewsContract.createAndSave(
-        {
-          title: dataViewName,
-        },
-        false,
-        true
-      );
+      const interval = setInterval(async () => {
+        // Ran out of retry attempts
+        if (retryCount === 0) {
+          clearInterval(interval);
+          addRequestMessage({
+            error: i18n.translate(
+              'xpack.ml.dataframe.analytics.create.destinationIndexNotCreatedForDataFrameAnalyticsJob',
+              {
+                defaultMessage: 'Destination index has not yet been created.',
+              }
+            ),
+            message: i18n.translate(
+              'xpack.ml.dataframe.analytics.create.unableToCreateDataViewForDataFrameAnalyticsJob',
+              {
+                defaultMessage: 'Unable to create data view.',
+              }
+            ),
+          });
+        }
 
-      addRequestMessage({
-        message: i18n.translate(
-          'xpack.ml.dataframe.analytics.create.createDataViewSuccessMessage',
-          {
-            defaultMessage: 'Kibana data view {dataViewName} created.',
-            values: { dataViewName },
+        try {
+          const resp = await ml.checkIndexExists(destinationIndex);
+          // Index exists - clear interval and create index pattern
+          if (resp && resp.exists) {
+            clearInterval(interval);
+            await mlContext.dataViewsContract.createAndSave(
+              {
+                title: dataViewName,
+              },
+              false,
+              true
+            );
+
+            addRequestMessage({
+              message: i18n.translate(
+                'xpack.ml.dataframe.analytics.create.createDataViewSuccessMessage',
+                {
+                  defaultMessage: 'Kibana data view {dataViewName} created.',
+                  values: { dataViewName },
+                }
+              ),
+            });
+          } else {
+            // Retry - index might not be created yet
+            retryCount -= 1;
           }
-        ),
-      });
+        } catch (e) {
```

Comment:
will the `catch` below (line 190) ever get hit? 
It looks like this catch will now catch everything.

Reply:
Updated so each try/catch catches relevant errors 5c5f24f7f64192367d2665034a6d914076cbd6cf